BinaryEncoding.md

```diff
@@ -263,8 +249,6 @@ a `data_segment` is:
 
 ### Name section
 
-ID: `name`
-
 The names section does not change execution semantics and a validation error in
 this section does not cause validation for the whole module to fail and is
 instead treated as if the section was absent. The expectation is that, when a
```

Comment:
We could, although I think the names section makes sense more broadly.  If we change the meaning from "user-defined" to "metadata" or "ignorable", then it wouldn't seem too out of place to have the core wasm spec define a metadata/ignorable section.